static/app/components/codecov/container/codecovParamsProvider.tsx

```diff
@@ -0,0 +1,72 @@
+import {useCallback, useEffect} from 'react';
+
+import type {CodecovContextData} from 'sentry/components/codecov/context/codecovContext';
+import {CodecovContext} from 'sentry/components/codecov/context/codecovContext';
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type CodecovProviderProps = {
+  children?: NonNullable<React.ReactNode>;
+};
+
+export default function CodecovQueryParamsProvider({children}: CodecovProviderProps) {
+  const organization = useOrganization();
+  const orgSlug = organization.slug;
+
+  const location = useLocation();
+  const [localStorageState, setLocalStorageState] = useLocalStorageState(
+    `codecov-selection:${orgSlug}`,
+    {} as Partial<CodecovContextData>
+  );
+  const navigate = useNavigate();
+
+  const updateSelectorData: CodecovContextData['updateSelectorData'] = useCallback(
+    data => {
+      const currentParams = new URLSearchParams(location.search);
+      currentParams.set(Object.keys(data)[0] as string, Object.values(data)[0] as string);
+      navigate(`${location.pathname}?${currentParams.toString()}`, {replace: true});
+    },
+    [location.search, location.pathname, navigate]
+  );
+
+  useEffect(() => {
+    const repository =
+      typeof location.query.repository === 'string' ? location.query.repository : null;
+    const integratedOrg =
+      typeof location.query.integratedOrg === 'string'
+        ? location.query.integratedOrg
+        : null;
+    const branch =
+      typeof location.query.branch === 'string' ? location.query.branch : null;
+    const codecovPeriod =
+      typeof location.query.codecovPeriod === 'string'
+        ? location.query.codecovPeriod
+        : null;
+
+    setLocalStorageState({repository, integratedOrg, branch, codecovPeriod});
+  }, [setLocalStorageState, location.query]);
+
+  const params: CodecovContextData = {
+    repository:
+      decodeURIComponent(location.query.repository as string).trim() ||
+      localStorageState.repository ||
+      null,
```

Comment:
Should there be any defaults for this? The way this reads right now is as if null is a valid value

Reply:
I was thinking of the nulls as a legitimate placeholder value until a person chooses a value for the selector. So if the value is null the selector would say like "choose repo" or something. Would you recommend against this and instead have the value be undefined or not sent via the context altogether?

Reply:
That makes sense, but I would make sure to document it. You are probably going to run into issues down the line where the fetch calls are being made with null values (unless you explicitly check obviously).

Reply:
Sounds good, left a comment explaining this 👌